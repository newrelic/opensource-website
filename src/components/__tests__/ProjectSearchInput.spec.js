import React from 'react';
import { render } from '@testing-library/react'; // https://testing-library.com/docs/react-testing-library/intro
// import { useStaticQuery } from 'gatsby';

import { get } from 'lodash';

import ProjectSearchInput from '../ProjectSearchInput';
import data from './fixtures/ProjectSearchInput';

const allLanguages = get(data, 'allLanguages.group', []);
const allCategories = get(data, 'allCategories.group', []);
const allProjectTags = get(data, 'allProjectTags.group', []);

const filterOptions = {
  allCategories: { title: 'Categories', options: allCategories },
  allLanguages: { title: 'Language', options: allLanguages },
  allProjectTags: { title: 'Type', options: allProjectTags },
};

describe('ProjectSearchInput', () => {
  it('Renders correctly', () => {
    const props = {
      filterOptions,
      filterValues: {
        ossCategory: '',
        projectTag: '',
        languageType: '',
      },
      searchQueryValue: '',
      onSearchQueryChange: () => {},
      onFilterChange: () => {},
    };
    const { container } = render(<ProjectSearchInput {...props} />);
    expect(container).toMatchSnapshot();
  });
});
