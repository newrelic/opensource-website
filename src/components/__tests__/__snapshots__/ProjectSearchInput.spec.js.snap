// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSearchInput Renders correctly 1`] = `
.emotion-0 {
  --horizontal-spacing: 1rem;
  position: relative;
  width: 100%;
  --icon-size: 1.5rem;
}

.emotion-1 {
  width: 1.25rem;
  height: 1.25rem;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  position: absolute;
  left: var(--horizontal-spacing);
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-2 {
  width: 100%;
  border: 1px solid var(--primary-text-color);
  border-radius: 4px;
  background: var(--secondary-background-color);
  -webkit-transition: 0.15s ease-out;
  transition: 0.15s ease-out;
  line-height: 1;
  color: var(--primary-text-color);
  padding-left: calc(
                      var(--horizontal-spacing) + 0.5rem + var(--icon-size)
                    );
  padding-right: var(--horizontal-spacing);
  font-size: 1.25rem;
  font-weight: 500;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.emotion-2:focus {
  outline: none;
  border: 1px solid rgba(0, 126, 138, 0.6);
  box-shadow: 0 0 0 4px rgba(0, 126, 138, 0.1);
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  position: relative;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 0.5rem 1rem;
  font-size: 0.875rem;
  font-weight: 400;
  border-radius: 3px;
  font-family: var(--primary-font-family);
  line-height: 1;
  cursor: pointer;
  border: 1px solid transparent;
  transition-duration: 0.15s;
  transition-property: color,background;
  transition-timing-function: ease-out;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--button-text-color);
  background-color: var(--button-background-color);
}

.emotion-4:hover {
  color: var(--button-text-color);
  background: var(--color-black);
}

.dark-mode .emotion-4:hover {
  color: var(--button-text-color);
  background: #d0d6d7;
}

.emotion-6 {
  width: 1em;
  height: 1em;
  fill: none;
  stroke: currentColor;
  stroke-width: 2;
  stroke-linecap: round;
  stroke-linejoin: round;
  -webkit-transform: rotate(0);
  -moz-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.emotion-7 {
  --arrow-size: 5px;
  --arrow-offset: 0.5rem;
  position: absolute;
  top: calc(100% + var(--arrow-size));
  display: none;
  background: var(--secondary-background-color);
  border-radius: 0.25rem;
  z-index: 1000;
  padding: 0.5rem;
  box-shadow: 0 3px 8px 0 rgba(22, 38, 59, 0.2);
}

.dark-mode .emotion-7 {
  background-color: var(--primary-hover-color);
}

.emotion-7::before {
  content: '';
  position: absolute;
  display: block;
  top: calc(-1 * var(--arrow-size));
  border-left: var(--arrow-size) solid transparent;
  border-right: var(--arrow-size) solid transparent;
  border-bottom: var(--arrow-size) solid var(--background-color);
  width: 0;
  height: 0;
  z-index: 1000;
  left: var(--arrow-offset);
}

.emotion-8 {
  max-height: 20rem;
  min-width: 100px;
  overflow-y: auto;
}

.emotion-9 {
  display: block;
  padding: 0.25rem 0.5rem;
  font-size: 0.75rem;
  -webkit-transition: all 0.2s ease-out;
  transition: all 0.2s ease-out;
  color: var(--text-color);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  color: var(--text-color);
  cursor: pointer;
  background: var(--primary-hover-color);
  border-radius: 0.25rem;
}

.dark-mode .emotion-9:hover {
  background-color: var(--system-border-regular-dark);
}

<div>
  <div>
    <div
      class="emotion-0"
    >
      <svg
        class="emotion-1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
      <input
        class="emotion-2"
        placeholder="Search for a project"
        type="text"
        value=""
      />
    </div>
    <div
      class="undefined undefined"
    >
      <span>
        Show filters
        <svg
          fill="none"
          height="12"
          stroke="#007e8a"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
          />
        </svg>
      </span>
      <div
        class="emotion-3"
      >
        <button
          class="emotion-4 emotion-5"
        >
          All Categories
          <svg
            class="emotion-6"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </button>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            <div
              class="emotion-9"
            >
              All Categories
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-3"
      >
        <button
          class="emotion-4 emotion-5"
        >
          All Project Tags
          <svg
            class="emotion-6"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </button>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            <div
              class="emotion-9"
            >
              All Project Tags
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-3"
      >
        <button
          class="emotion-4 emotion-5"
        >
          All Language Types
          <svg
            class="emotion-6"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              points="6 9 12 15 18 9"
            />
          </svg>
        </button>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-8"
          >
            <div
              class="emotion-9"
            >
              All Language Types
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
